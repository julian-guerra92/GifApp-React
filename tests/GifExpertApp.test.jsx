const { render, screen, fireEvent } = require("@testing-library/react")
const { GifExpertApp } = require("../src/GifExpertApp")

describe('Pruebas en <GifExpertApp/> ', () => {

   test('Debe de hacer math con el snapshot', () => {
      const { container } = render(<GifExpertApp />);
      expect(container).toMatchSnapshot();
   })

   test('Debe de mostrar los textos "One Punch" y "Dragon Ball"', () => {
      render(<GifExpertApp />);
      expect(screen.getByText('One Punch'));
      expect(screen.getByText('Dragon Ball'));
   })

   test('Se debe agregar una categorÃ­a si se ingresa una diferente a "One Punch" y "Dragon Ball"', () => {
      render(<GifExpertApp />);
      const input = screen.getByRole('textbox');
      const form = screen.getByRole('form');
      fireEvent.input(input, { target: { value: 'Witcher' } });
      fireEvent.submit(form);
      expect(screen.getByText('Witcher'));
   })

})