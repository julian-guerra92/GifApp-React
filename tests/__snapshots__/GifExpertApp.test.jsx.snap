// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifExpertApp/>  Debe de hacer math con el snapshot 1`] = `
<div>
  <h1>
    GifExpertApp
  </h1>
  <form
    aria-label="form"
  >
    <input
      placeholder="Buscar Gifts"
      type="text"
      value=""
    />
  </form>
  <h3>
    One Punch
  </h3>
  <h2>
    Cargando...
  </h2>
  <div
    class="card-grid"
  />
  <h3>
    Dragon Ball
  </h3>
  <h2>
    Cargando...
  </h2>
  <div
    class="card-grid"
  />
</div>
`;
